cmd/podman/ps.go

```diff
@@ -693,3 +631,77 @@ func getPorts(ports []ocicni.PortMapping) string {
 	}
 	return strings.Join(portDisplay, ", ")
 }
+
+type batchContainerStruct struct {
+	conConfig          *libpod.ContainerConfig
+	conState           libpod.ContainerStatus
+	exitCode           int32
+	pid                int
+	rootFsSize, rwSize int64
+	startedTime        time.Time
+}
+
+func batchContainerOp(ctr *libpod.Container, opts psOptions) (batchContainerStruct, error) {
+	var (
+		conConfig          *libpod.ContainerConfig
+		conState           libpod.ContainerStatus
+		err                error
+		exitCode           int32
+		pid                int
+		rootFsSize, rwSize int64
+		startedTime        time.Time
+	)
+
+	batchErr := ctr.Batch(func(c *libpod.Container) error {
+		conConfig = c.Config()
+		conState, err = c.State()
+		if err != nil {
+			return errors.Wrapf(err, "unable to obtain container state")
+		}
+
+		exitCode, err = c.ExitCode()
+		if err != nil {
+			return errors.Wrapf(err, "unable to obtain container exit code")
+		}
+		startedTime, err = c.StartedTime()
+		if err != nil {
+			logrus.Errorf("error getting started time for %q: %v", c.ID(), err)
+		}
+
+		if !opts.size && !opts.namespace {
```

Comment:
This should be at top.

Reply:
I think this is intentional - he's exiting early if he doesn't need to get those two things

Reply:
Right but this should happen before he makes the above function calls.  Since the functions don't effect this at all.